client/batch.go

```diff
@@ -308,6 +315,29 @@ func (b *Batch) Scan(s, e interface{}, maxRows int64) {
 	b.initResult(1, 0, nil)
 }
 
+// ReverseScan retrieves the rows in reserve order starting at end (exclusive)
+// and finishing at begin(inclusive).
+//
+// A new result will be appended to the batch which will contain up to maxRows
+// rows and Result.Err will indicate success or failure.
+//
+// key can be either a byte slice, a string, a fmt.Stringer or an
+// encoding.BinaryMarshaler.
+func (b *Batch) ReverseScan(s, e interface{}, maxRows int64) {
```

Comment:
It would definitely have some not-so-subtle issues, namely that it requires an update to our routing. Probably not a good idea internally, but for client usage could still make sense - but that's less important now.

An unrelated thought though: Some of the complexity seems to stem from the fact that with a `ReverseScan` on `"a"`-`"z"`, the `EndKey` `"z"` is inclusive. If we implemented only `EndKey`-exclusive `ReverseScan` internally, that would be easier (because it plays along with `EndKey` being excluded in range operations) and we can easily expose the inclusive one by calling `Next()` on the EndKey for clients. Have you considered that?